pkg/logs/tailers/windowsevent/transform.go

```diff
@@ -0,0 +1,208 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build windows
+
+package windowsevent
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	"github.com/clbanning/mxj"
+	"golang.org/x/text/encoding/unicode"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// eventToJSON converts an XML message into json
+func eventToJSON(re *richEvent) (string, error) {
+	event := re.xmlEvent
+	log.Trace("Rendered XML:", event)
+	mxj.PrependAttrWithHyphen(false)
+	mv, err := mxj.NewMapXml([]byte(event))
+	if err != nil {
+		return "", err
+	}
+
+	// extract then modify the Event.EventData.Data field to have a key value mapping
+	err = formatEventDataField(mv)
+	if err != nil {
+		log.Debugf("Error formatting %s: %s", dataPath, err)
```

Comment:
It needs to be `%s` or `%v` here, since its a log line and not a `fmt.Errorf`, but I replaced those in https://github.com/DataDog/datadog-agent/pull/19761/commits/51ce96132101ec7999ab1eb485d91ee5285e2404